### ISSUE TITLE
Add Module Namespace Exotic Objects Not extensible Lint Rule

### ISSUE THREAD

**Issue by BARMPlus** [eslint/eslint#14911]
**Please describe what the rule should do:**

Module namespace exotic objects are not extensible according to the ECMAScript spec,But now eslint is not check。

**What new ECMAScript feature does this rule relate to?**

 ```Module Namespace Exotic Objects```

[TC39 - Module Namespace Exotic Objects](https://tc39.es/ecma262/#sec-module-namespace-exotic-objects)

![image](https://user-images.githubusercontent.com/14230248/128841081-f1143e6b-d909-44ba-8acf-714084d5c0f0.png)

**What category of rule is this? (place an "X" next to just one item)**

- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

[Demo Address](https://github.com/BARMPlus/eslint-rule-module-namespace-exotic-objects)
```js
import * as consts from './consts'

consts.bar = 'bar' // error here

consts.foo = function () { // error here

}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
This error is a javascript logic error and is more suitable for eslint。

**Are you willing to submit a pull request to implement this rule?**
If I have time and ability, I am willing。

**Comment by aladdin-add** [comment]
should also warns: 
```js
import * as consts from './consts';

Object.assign(consts, {bar: 'bar'}); // runtime error.
```

👍 I will support it. (and, I was the one suggested filing the issue :))

**Comment by mdjermanovic** [comment]
This is already covered by the [no-import-assign](https://eslint.org/docs/rules/no-import-assign) rule.

```js
/* eslint no-import-assign: "error" */

import * as consts from './consts'

consts.bar = 'bar' // error

consts.foo = function () {} // error

Object.assign(consts, {bar: 'bar'}); // error
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcG9ydC1hc3NpZ246IFwiZXJyb3JcIiAqL1xuXG5pbXBvcnQgKiBhcyBjb25zdHMgZnJvbSAnLi9jb25zdHMnXG5cbmNvbnN0cy5iYXIgPSAnYmFyJyAvLyBlcnJvclxuXG5jb25zdHMuZm9vID0gZnVuY3Rpb24gKCkge30gLy8gZXJyb3JcblxuT2JqZWN0LmFzc2lnbihjb25zdHMsIHtiYXI6ICdiYXInfSk7IC8vIGVycm9yXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

**Comment by BARMPlus** [comment]
> This is already covered by the [no-import-assign](https://eslint.org/docs/rules/no-import-assign) rule.
> 
> 这已经包含在禁止导入分配规则中了。
> 
> ```js
> /* eslint no-import-assign: "error" */
> 
> import * as consts from './consts'
> 
> consts.bar = 'bar' // error
> 
> consts.foo = function () {} // error
> 
> Object.assign(consts, {bar: 'bar'}); // error
> ```
> 
> [Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWltcG9ydC1hc3NpZ246IFwiZXJyb3JcIiAqL1xuXG5pbXBvcnQgKiBhcyBjb25zdHMgZnJvbSAnLi9jb25zdHMnXG5cbmNvbnN0cy5iYXIgPSAnYmFyJyAvLyBlcnJvclxuXG5jb25zdHMuZm9vID0gZnVuY3Rpb24gKCkge30gLy8gZXJyb3JcblxuT2JqZWN0LmFzc2lnbihjb25zdHMsIHtiYXI6ICdiYXInfSk7IC8vIGVycm9yXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)
> 
> 在线演示

Yes i will close this question
